cloudify_azure/resources/storage/disk.py

```diff
@@ -0,0 +1,147 @@
+        exists or None if it does not
+    '''
+    ctx.logger.debug('Checking if Data Disk "{0}/{1}" exists'
+                     .format(disk_container, disk_name))
+    try:
+        props = pageblobsvc.get_blob_properties(disk_container, disk_name)
+        ctx.logger.debug('Data Disk "{0}/{1}" exists'
+                         .format(disk_container, disk_name))
+        return props
+    except Exception:
+        ctx.logger.debug('Data Disk "{0}/{1}" does not exist'
+                         .format(disk_container, disk_name))
+        return None
+
+
+def get_cloud_storage_account(_ctx=ctx):
+    '''Gets the Azure Blob storage service'''
+    # Get the storage account
+    storage_account = utils.get_parent(
+        _ctx.instance,
+        rel_type=constants.REL_CONTAINED_IN_SA)
+    storage_account_name = utils.get_resource_name(_ctx=storage_account)
+    # Get the storage account keys
+    keys = StorageAccount(_ctx=storage_account).list_keys()
+    if not isinstance(keys, list) or len(keys) < 1:
+        raise RecoverableError(
+            'StorageAccount reported no usable authentication keys')
+    # Get an interface to the Storage Account
+    storage_account_key = keys[0].get('key')
+    return CloudStorageAccount(
+        account_name=storage_account_name,
+        account_key=storage_account_key)
+
+
+@operation
+def create_data_disk(**_):
+    '''Uses an existing, or creates a new, Data Disk placeholder'''
+    res_cfg = utils.get_resource_config() or dict()
+    disk_name = ctx.node.properties.get('name')
+    disk_container = res_cfg.get('container_name')
+    # Validation
+    if ctx.node.properties.get('use_external_resource', False):
+        if not disk_name:
+            raise NonRecoverableError(
+                '"use_external_resource" specified without '
+                'a resource "name"')
+        if not disk_container:
+            raise NonRecoverableError(
+                '"use_external_resource" specified without '
+                'a resource "container_name"')
+    # Get the storage account
+    csa = get_cloud_storage_account()
+    # Get an interface to the Page Blob Service
+    pageblobsvc = csa.create_page_blob_service()
+    # Generate a VHD Data Disk name if needed
+    if not disk_name:
+        ctx.logger.info('Generating a new Data Disk name')
+        for _ in xrange(0, 10):
+            tmpname = disk_name_generator()
+            if not data_disk_exists(pageblobsvc, disk_container, tmpname):
+                disk_name = tmpname
+                break
+    # Set the runtime properties
+    ctx.instance.runtime_properties['name'] = disk_name
+    ctx.instance.runtime_properties['diskSizeGB'] = \
+        res_cfg.get('size')
+    ctx.instance.runtime_properties['container'] = \
+        disk_container
+    ctx.instance.runtime_properties['uri'] = \
+        constants.CONN_STORAGE_BLOB_ENDPOINT.format(
+            csa.account_name) + '/{0}/{1}'.format(disk_container, disk_name)
+
```

Comment:
In the rest of the plugin, you've normally sent runtime properties to debug log at the end of a create method.